modules/k6/k6.go

```diff
@@ -17,17 +21,78 @@ type K6Container struct {
 	testcontainers.Container
 }
 
+type RemoteTestFileDescription struct{
+
+	Uri string
+	DownloadDir string
+	User string
+	Password string
+}
+
+func ( d *RemoteTestFileDescription) getDownloadPath() string{
+	baseName := path.Base(d.Uri)
+	return path.Join(d.DownloadDir,baseName)
+
+}
+
+func downloadFileFromDescription(d RemoteTestFileDescription) error{
+
+	client := http.Client{Timeout: time.Second*60}
+	// Set up HTTPS request with basic authorization.
+	req, err := http.NewRequest(http.MethodGet, d.Uri, nil)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Content-Type", "text/javascript")
+	if d.User != "" && d.Password != ""{
+		req.SetBasicAuth(d.User, d.Password)
+	}	
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	
+
+	out, err := os.Create(d.getDownloadPath())
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	// Write the body to file
+	_, err = io.Copy(out, resp.Body)
+	return err
+
+
+
+}
+
+
 // WithTestScript mounts the given script into the ./test directory in the container
 // and passes it to k6 as the test to run.
 // The path to the script must be an absolute path
 func WithTestScript(scriptPath string) testcontainers.CustomizeRequestOption {
-	return func(req *testcontainers.GenericContainerRequest) {
-		script := filepath.Base(scriptPath)
-		target := "/home/k6x/" + script
+	
+		scriptBaseName := filepath.Base(scriptPath)
+		f, err := os.Open(scriptPath)
+		if err != nil {
+			panic("Cannot create reader for test file ")
+		}
+		return WithTestScriptReader(f,scriptBaseName)
+	
+}
+
+// Copies files into the Container using the Reader API
```

Comment:
Could you follow the Go conventions for docs? The function comments should start with the name of the function.